lib/eclipse/CMakeLists.txt

```diff
@@ -24,21 +24,18 @@ add_executable(genkw Parser/createDefaultKeywordList.cpp
 target_link_libraries(genkw opmjson ecl boost_regex)
 target_include_directories(genkw PRIVATE include)
 
-file(GLOB_RECURSE keyword_templates share/keywords/*)
+include( share/keywords/keyword_list.cmake )
+string(REGEX REPLACE "([^;]+)" "${CMAKE_CURRENT_SOURCE_DIR}/share/keywords/\\1" keyword_files "${keywords}")
+
 add_custom_command(
-    OUTPUT  ${CMAKE_CURRENT_BINARY_DIR}/ParserKeywords.cpp
-    COMMAND genkw ${PROJECT_SOURCE_DIR}/share/keywords
-                  ${CMAKE_CURRENT_BINARY_DIR}/ParserKeywords.cpp
+    OUTPUT  ${CMAKE_CURRENT_BINARY_DIR}/ParserKeywords.cpp  ${CMAKE_CURRENT_BINARY_DIR}/inlinekw.cpp
+
+    COMMAND genkw ${CMAKE_CURRENT_BINARY_DIR}/ParserKeywords.cpp
                   ${CMAKE_CURRENT_BINARY_DIR}/include/
                   opm/parser/eclipse/Parser/ParserKeywords
-    DEPENDS genkw ${keyword_templates}
-)
-
-add_custom_command(
-    OUTPUT  ${CMAKE_CURRENT_BINARY_DIR}/inlinekw.cpp
-    COMMAND genkw ${PROJECT_SOURCE_DIR}/share/keywords
                   ${CMAKE_CURRENT_BINARY_DIR}/inlinekw.cpp
-    DEPENDS genkw ${keyword_templates}
+                  ${keyword_files}
+    DEPENDS genkw ${keyword_files} share/keywords/keyword_list.cmake
```

Comment:
I agree that when you use `include()` you have a dependency to the *full*  cmake system, i.e. changing the keyword file will give a full reconfigure. However that reconfigure will not rerun the custom command which creates keyword files.

I have tested this - but might have stumbled in the testing.